### PR TITLE
test configs: Allow ltp-timers to be tested in qemu device

### PR DESCRIPTION
**Pull request by lubinszARM**
Signed-off-by: Robin Lu <robin.lubin@shopee.com>

### CODE REVIEW

**Comment by lubinszARM** [comment]
Hi @gctucker 
I have deployed qemu+ltp/kselftest related tests in my local kernelci+lava environment and it has been running stably for a period of time.
Also, please see the log of qemu+ltp as reference.
[lava_qemu_ltp_timers.log](https://github.com/kernelci/kernelci-core/files/7691823/lava_qemu_ltp_timers.log)

**Comment by lubinszARM** [comment]
cc @broonie

File: config/core/test-configs.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ file_systems:
 
   debian_buster-ltp_nfs:
     type: debian
-    ramdisk: 'buster-ltp/20211203.0/{arch}/initrd.cpio.gz'
+    ramdisk: 'buster-ltp/20211203.0/{arch}/rootfs.cpio.gz'
```
Comment (gctucker) [inline config/core/test-configs.yaml:87]: The `initrd.cpio.gz` is created especially to then be able to switch the root file system to NFS.  So this change can't be done, the `rootfs.cpio.gz` is to boot with just a ramdisk which has a full root file system.

We either need to fix NFS root on QEMU or make a variant for LTP tests to run on QEMU with a ramdisk.

File: config/core/test-configs.yaml [hunk 2]
```diff
@@ -1342,6 +1342,7 @@ device_types:
     context:
       arch: x86_64
       cpu: 'qemu64'
+      memory: 1024
```
Comment (gctucker) [inline config/core/test-configs.yaml:1345]: Why was this needed?

File: config/core/test-configs.yaml (hunk 1 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml:87]: I see. I will try to add a new variant for this case.

File: config/core/test-configs.yaml (hunk 2 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml]: The default 512M seems too small. When running ltp/kselftest+ qemu related cases, there will be random OOM crash problems. Especially in ltp-mm, kselftest-lib ...

File: config/core/test-configs.yaml (hunk 1 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml]: Hi @gctucker 
I modified a new version according to your suggestion.
Please help to check it.
Thanks.

File: config/core/test-configs.yaml (hunk 2 above)
Comment (gctucker) [reply config/core/test-configs.yaml]: OK thanks, let's see if we can have ltp-timers working with the default memory size on QEMU and then if it's needed for other tests we can potentially increase it.

One issue is that some labs may be configured for a certain maximum amount of memory per QEMU instance, so changing the memory attribute may not be enough if there isn't enough actual physical RAM on the machines where the jobs are run.  We might need to add an attribute in the test plan definitions about the amount of memory required so labs can enable only those that they will be able to run.  But that would be a discussion for a follow-up PR I think.

File: config/core/test-configs.yaml [hunk 3]
```diff
@@ -336,6 +339,15 @@ test_plans:
       grp_test: "TMR"
       job_timeout: '30'
 
+  ltp-timers-qemu:
+    <<: *ltp
+    rootfs: debian_buster-ltp_qemu
```
Comment (gctucker) [inline config/core/test-configs.yaml:344]: To group the results under ltp-timers, with the tests run on non-QEMU devices:
```suggestion
  ltp-timers_qemu:
    <<: *ltp
    basename: ltp-timers
    rootfs: debian_buster-ltp_qemu
```

**Review by gctucker — COMMENTED** [review]
Tested OK on staging, except that it didn't work at all in lab-broonie for some reason:
https://staging.kernelci.org/test/job/kernelci/branch/staging-mainline/kernel/staging-mainline-20211214.0/plan/ltp-timers-qemu/

@broonie Do you know why these failed only in your lab?

It worked in the Collabora and LKFT labs although there were a few test case failures in all the jobs.  It would be good to see if these can be fixed (missing config?) or if they should be skipped.

File: config/core/test-configs.yaml [hunk 4]
```diff
@@ -339,8 +339,9 @@ test_plans:
       grp_test: "TMR"
       job_timeout: '30'
 
-  ltp-timers-qemu:
+  ltp-timers_qemu:
```
Comment (lubinszARM) [inline config/core/test-configs.yaml:342]: In the test plan is still ltp-timers-qemu, we also need to change it.

Comment (gctucker) [reply config/core/test-configs.yaml:342]: Sure.  Actually I think this fix can be squashed with the first commit.

**Comment by gctucker** [comment]
Adding `staging-skip` label as there's a YAML validation error as anticipated in a comment:
```
KeyError: 'ltp-timers-qemu'
```

File: config/core/test-configs.yaml (hunk 4 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml:342]: Done.

**Comment by lubinszARM** [comment]
@gctucker 
Any progress? Can we merge it?

**Comment by gctucker** [comment]
I believe the change in the rootfs images hasn't been tested yet, with the symlink stage moved to a different place.

**Comment by lubinszARM** [comment]
> symlink

Oh. I see.
But https://github.com/kernelci/kernelci-core/pull/929 has nothing to do with this PR.
full.rootfs.cpio.gz is only required by kselftest@qemu related cases.
For ltp@qemu, rootfs.cpio.gz can already meet the requirements.

File: config/core/test-configs.yaml [hunk 5]
```diff
@@ -88,6 +88,9 @@ file_systems:
     nfs: 'buster-ltp/20211203.0/{arch}/full.rootfs.tar.xz'
     root_type: nfs
 
+  debian_buster-ltp_qemu:
```
Comment (gctucker) [inline config/core/test-configs.yaml:91]:
```suggestion
  debian_buster-ltp_ramdisk:
```

File: config/core/test-configs.yaml [hunk 6]
```diff
@@ -336,6 +339,16 @@ test_plans:
       grp_test: "TMR"
       job_timeout: '30'
 
+  ltp-timers_qemu:
+    <<: *ltp
+    basename: ltp-timers
+    rootfs: debian_buster-ltp_qemu
```
Comment (gctucker) [inline config/core/test-configs.yaml:345]:
```suggestion
    rootfs: debian_buster-ltp_ramdisk
```

File: config/core/test-configs.yaml [hunk 7]
```diff
@@ -336,6 +339,16 @@ test_plans:
       grp_test: "TMR"
       job_timeout: '30'
 
+  ltp-timers_qemu:
+    <<: *ltp
+    basename: ltp-timers
```
Comment (gctucker) [inline config/core/test-configs.yaml:344]:
```suggestion
    base_name: ltp-timers
```

**Review by gctucker — COMMENTED** [review]
The tests were run fine:
https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20211220.0/plan/ltp-timers_qemu/

A couple of small tweaks are still needed then this should be ready to be merged.

**Comment by lubinszARM** [comment]
@gctucker 
Sorry for delay due to the holiday. Please check it again. :)

File: config/core/test-configs.yaml [hunk 8]
```diff
@@ -88,7 +88,9 @@ file_systems:
     type: debian
     ramdisk: 'buster-v4l2/20211224.3/{arch}/rootfs.cpio.gz'
 
-
+  debian_buster-ltp_ramdisk:
```
Comment (gctucker) [inline config/core/test-configs.yaml:91]: We've moved on to Bullseye now for this rootfs, so please use `debian_bullseye-ltp_ramdisk` and move it after the `-nfs` variant.

File: config/core/test-configs.yaml [hunk 9]
```diff
@@ -356,6 +358,16 @@ test_plans:
       grp_test: "TMR"
       job_timeout: '30'
 
+  ltp-timers_qemu:
+    <<: *ltp
+    base_name: ltp-timers
+    rootfs: debian_buster-ltp_ramdisk
```
Comment (gctucker) [inline config/core/test-configs.yaml:364]:
```suggestion
    rootfs: debian_bullseye-ltp_ramdisk
```

File: config/core/test-configs.yaml (hunk 8 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml]: Done.

File: config/core/test-configs.yaml (hunk 9 above)
Comment (lubinszARM) [reply config/core/test-configs.yaml]: Done.

**Review by gctucker — APPROVED** [review]
This one is failing to boot in @broonie's lab:
https://staging.kernelci.org/test/plan/id/61d4f37a573563a138635574/

But otherwise results seem valid in the other labs:
https://staging.kernelci.org/test/plan/id/61d4e8c5b30a9bbea1635589/ (gcc-10)
https://staging.kernelci.org/test/plan/id/61d500d188931996d763555c/ (clang-11)

So I think this PR can be merged but we should take a look at what's causing the failure.

**Comment by gctucker** [comment]
The commits also need to be squashed or reorganised.

**Comment by gctucker** [comment]
@lubinszARM Please rebase and squash fixups

**Comment by lubinszARM** [comment]
> @lubinszARM Please rebase and squash fixups

Hi
Done.

**Comment by lubinszARM** [comment]
Any comments?

**Review by gctucker — APPROVED** [review]
Tested OK on staging:
https://staging.kernelci.org/test/plan/id/61f17f2f5753e3db1338decf/

It's failing to boot in @broonie's lab but that's a separate issue I believe:
https://staging.kernelci.org/test/plan/id/61f1794cc1032fb06d38de22/
